File: src/main/java/no/nav/samordning/hendelser/feed/Feed.java
package no.nav.samordning.hendelser.feed;

import no.nav.samordning.hendelser.hendelse.Hendelse;

import java.util.List;

public class Feed {

    private List<Hendelse> hendelser;
    private String nextUrl;

    public Feed(List<Hendelse> hendelser, String nextUrl) {
        this.hendelser = hendelser;
        this.nextUrl = nextUrl;
    }

    public List<Hendelse> getHendelser() {
        return hendelser;
    }

    public void setHendelser(List<Hendelse> hendelser) {
        this.hendelser = hendelser;
    }

    public String getNextUrl() {
        return nextUrl;
    }

    public void setNextUrl(String nextUrl) {
        this.nextUrl = nextUrl;
    }
}

File: src/main/kotlin/no/nav/samordning/hendelser/common/feed/Feed.kt
package no.nav.samordning.hendelser.common.feed

import org.springframework.data.domain.Page
import java.net.URI

data class Feed<T: SequentialDTO>(var hendelser: List<T>, var sisteSekvensnummer: Long, var sisteLesteSekvensnummer: Long, var nextUrl: URI?) {

    constructor(page: Page<T>, queryURL: URI?) : this(
        page.content,
        page.totalElements,
        page.maxOfOrNull { it.sekvensnummer } ?: 0,
        queryURL
    )
}
